include/open62541/plugin/nodestore.h

```diff
@@ -411,6 +411,12 @@ struct UA_NodeHead {
     UA_MonitoredItem *monitoredItems; /* MonitoredItems for Events and immediate
                                        * DataChanges (no sampling interval). */
 #endif
+#ifdef UA_ENABLE_ROLE_PERMISSION
+    size_t rolePermissionsSize;
+    UA_RolePermissionType *rolePermissions;
+    size_t userRolePermissionsSize;
```

Comment:
The user roles are not static but depend on the session that is reading the node.
Do not store the userRolePermissions as an explicit field in the nodes.

Rather modify the "Read" service to compute this value at runtime.
Just return all RolePermissions where both the role is defined for the node **and** the current session has that role.